File: backend/src/main/java/com/ll/nbe342team8/domain/qna/answer/dto/AnswerDto.java
package com.ll.nbe342team8.domain.qna.answer.dto;

import com.ll.nbe342team8.domain.qna.answer.entity.Answer;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;

public record AnswerDto(
        Long id,
        LocalDateTime createDate,
        LocalDateTime modifyDate,
        @NotNull String content
) {
    public AnswerDto(Answer answer) {
        this(
                answer.getId(),
                answer.getCreateDate(),
                answer.getModifyDate(),
                answer.getContent()
        );
    }
}

File: backend/src/main/java/com/ll/nbe342team8/domain/qna/answer/dto/AnswerDto.kt
package com.ll.nbe342team8.domain.qna.answer.dto

import com.ll.nbe342team8.domain.qna.answer.entity.Answer
import java.time.LocalDateTime

data class AnswerDto(
    val id: Long,
    val createDate: LocalDateTime,
    val modifyDate: LocalDateTime,
    val content: String
) {
    constructor(answer: Answer) : this(
        id = answer.id,
        createDate = answer.createDate,
        modifyDate = answer.modifyDate,
        content = answer.content
    )
}
